modules/basegl/include/modules/basegl/processors/geometryentryexitpoints.h

```diff
@@ -0,0 +1,91 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <modules/basegl/baseglmoduledefine.h>
+#include <inviwo/core/common/inviwo.h>
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/properties/boolproperty.h>
+#include <inviwo/core/properties/cameraproperty.h>
+#include <inviwo/core/ports/imageport.h>
+#include <inviwo/core/ports/volumeport.h>
+#include <inviwo/core/ports/meshport.h>
+#include <inviwo/core/interaction/cameratrackball.h>
+
+#include <modules/basegl/algorithm/entryexitpoints.h>
+
+namespace inviwo {
+
+/** \docpage{org.inviwo.GeometryEntryExitPoints, Geometry Entry Exit Points}
+ * ![](org.inviwo.GeometryEntryExitPoints.png?classIdentifier=org.inviwo.GeometryEntryExitPoints)
+ * Computes entry and exit points of a given mesh based on the current camera parameters.
+ * The positions of the input geometry are mapped to Data space, i.e. texture coordinates, of the
+ * input volume. The output color will be zero if no intersection is found, otherwise.
+ *
+ * ### Inports
+ *   * __volume__     Input volume used to map geometry positions to Data space
+ *   * __geometry__   Input geometry used for determining entry and exit points
+ *
+ * ### Outports
+ *   * __entry__ The first hit point in texture coordinates [0,1]
+ *   * __exit__  The last hit point in texture coordinates [0,1]
+ *
+ * ### Properties
+ *   * __Camera__   Scene camera
+ */
+
+/**
+ * \brief VERY_BRIEFLY_DESCRIBE_THE_PROCESSOR
+ * DESCRIBE_THE_PROCESSOR_FROM_A_DEVELOPER_PERSPECTIVE
+ */
+class IVW_MODULE_BASEGL_API GeometryEntryExitPoints : public Processor {
```

Comment:
Should this not be MeshEntryExitPoints?

Reply:
that's the big question. We use geometry and mesh somewhat interchangeably for some reason...

Reply:
I think that we started with Geometry as a general term for triangular meshes, point clouds, procedural geometry, etc. 
Then, we more or less only supported Meshes so we renamed the geometry to Mesh. Hence, we do not have a geometry class anymore.